File: PAT (Basic-Level)/C/1018.c
#include <stdio.h>
#include <stdlib.h>

int main(int argc, char const *argv[])
{
    int N;
    scanf("%d", &N);
    char A, B;
    int AWinTimes = 0, AFailTimes = 0, AEqualTimes = 0;
    int AWinByC = 0, AWinByJ = 0, AWinByB = 0;
    int BWinByC = 0, BWinByJ = 0, BWinByB = 0;
    getchar(); //clear the '\n'
    for (int i = 0; i < N; ++i)
    {
        scanf("%c %c", &A, &B);
        getchar();//clear the '\n'
        if((A == 'C' && B == 'J') || (A == 'J' && B == 'B') || (A == 'B' && B == 'C')){
            AWinTimes++;
            if(A == 'C')
                AWinByC++;
            if(A == 'J')
                AWinByJ++;
            if(A == 'B')
                AWinByB++;
        }
        else if(A == B)
            AEqualTimes++;
        else{
            AFailTimes++;
            if(A == 'C')
                BWinByB++;
            if(A == 'J')
                BWinByC++;
            if(A == 'B')
                BWinByJ++;
        }
    }
    printf("%d %d %d\n", AWinTimes, AEqualTimes, AFailTimes);
    printf("%d %d %d\n", AFailTimes, AEqualTimes, AWinTimes);
    if((AWinByB >= AWinByC) && (AWinByB >= AWinByJ))
        printf("B");
    else if(AWinByC >= AWinByJ)
        printf("C");
    else
        printf("J");

    if((BWinByB >= BWinByC) && (BWinByB >= BWinByJ))
        printf(" B");
    else if(BWinByC >= BWinByJ)
        printf(" C");
    else
        printf(" J");
    return 0;
}

File: PAT (Basic-Level)/Python/1018.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

N = int(input())
AWinTimes = 0; AFailTimes = 0; AEqualTimes = 0
AWinByC = 0; AWinByJ = 0; AWinByB = 0
BWinByC = 0; BWinByJ = 0; BWinByB = 0
for i in range(N):
    temp = input().split()
    if (temp[0] == 'C' and temp[1] == 'J') or (temp[0] == 'J' and temp[1] == 'B') or (temp[0] == 'B' and temp[1] == 'C'):
        AWinTimes += 1
        if temp[0] == 'C':
            AWinByC += 1
        if temp[0] == 'J':
            AWinByJ += 1
        if temp[0] == 'B':
            AWinByB += 1
    elif temp[0] == temp[1]:
        AEqualTimes += 1
    else:
        AFailTimes += 1
        if temp[0] == 'C':
            BWinByB += 1
        if temp[0] == 'J':
            BWinByC += 1
        if temp[0] == 'B':
            BWinByJ += 1

print("%d %d %d" % (AWinTimes, AEqualTimes, AFailTimes))
print("%d %d %d" % (AFailTimes, AEqualTimes, AWinTimes))

if AWinByB >= AWinByC and AWinByB >= AWinByJ:
    print("B", end = '')
elif AWinByC >= AWinByJ:
    print("C", end = '')
else:
    print("J", end = '')

if BWinByB >= BWinByC and BWinByB >= BWinByJ:
    print(" B", end = '')
elif BWinByC >= BWinByJ:
    print(" C", end = '')
else:
    print(" J", end = '')

#last testpoint overtime